starknet_py/net/account/base_account.py

```diff
@@ -0,0 +1,173 @@
+        :return: Signed Declare transaction
+        """
+
+    @abstractmethod
+    async def sign_deploy_account_transaction(
+        self,
+        class_hash: int,
+        contract_address_salt: int,
+        constructor_calldata: Optional[List[int]] = None,
+        *,
+        max_fee: Optional[int] = None,
+        auto_estimate: bool = False,
+    ) -> DeployAccount:
+        """
+        Create and sign deploy account transaction
+
+        :param class_hash: Class hash of the contract class to be deployed
+        :param contract_address_salt: A salt used to calculate deployed contract address
+        :param constructor_calldata: Calldata to be ed to contract constructor
+            and used to calculate deployed contract address
+        :param max_fee: Max fee to be paid for deploying account transaction. Enough tokens must be prefunded before
+            sending the transaction for it to succeed.
+        :param auto_estimate: Use automatic fee estimation, not recommend as it may lead to high costs
+        :return: Signed DeployAccount transaction
+        """
+
+    @abstractmethod
+    async def execute(
+        self,
+        calls: Calls,
+        *,
+        max_fee: Optional[int] = None,
+        auto_estimate: bool = False,
+        # TODO remove this
+        **kwargs,
+    ) -> SentTransactionResponse:
+        """
+        Takes calls and executes transaction
+
+        :param calls: Single call or list of calls
+        :param max_fee: Max amount of Wei to be paid when executing transaction
+        :param auto_estimate: Use automatic fee estimation, not recommend as it may lead to high costs
+        :return: SentTransactionResponse
+        """
+
+    @abstractmethod
+    def sign_message(self, typed_data: TypedData) -> List[int]:
+        """
+        Sign an TypedData TypedDict for off-chain usage with the starknet private key and return the signature
+        This adds a message prefix, so it can't be interchanged with transactions
+
+        :param typed_data: TypedData TypedDict to be signed
+        :return: The signature of the TypedData TypedDict
+        """
+
+    @abstractmethod
+    def hash_message(self, typed_data: TypedData) -> int:
+        """
+        Hash a TypedData TypedDict with pedersen hash and return the hash
+        This adds a message prefix, so it can't be interchanged with transactions
+
+        :param typed_data: TypedData TypedDict to be hashed
+        :return: the hash of the TypedData TypedDict
+        """
+
+    @abstractmethod
+    async def verify_message(self, typed_data: TypedData, signature: List[int]) -> bool:
+        """
+        Verify a signature of a TypedData TypedDict
+
+        :param typed_data: TypedData TypedDict to be verified
+        :param signature: signature of the TypedData TypedDict
+        :return: true if the signature is valid, false otherwise
+        """
```

Comment:
Does all that need to be in the account? 🤔

Reply:
(use cases for these are quite rare)

Reply:
A TypedData module or something else. I just don't think it is useful here.

Reply:
I think it can be done

Reply:
I think only hash_message can be removed, it is not related to the account.